terragrunt/aws/cloudfront/waf.tf

```diff
@@ -0,0 +1,334 @@
+        vendor_name = "AWS"
+
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "AWSManagedRulesKnownBadInputsRuleSet"
+      sampled_requests_enabled   = true
+    }
+  }
+
+  rule {
+    name     = "AWSManagedRulesLinuxRuleSet"
+    priority = 50
+
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesLinuxRuleSet"
+        vendor_name = "AWS"
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "AWSManagedRulesLinuxRuleSet"
+      sampled_requests_enabled   = true
+    }
+  }
+
+  rule {
+    name     = "AWSManagedRulesSQLiRuleSet"
+    priority = 60
+
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesSQLiRuleSet"
+        vendor_name = "AWS"
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "AWSManagedRulesSQLiRuleSet"
+      sampled_requests_enabled   = true
+    }
+  }
+
+  visibility_config {
+    cloudwatch_metrics_enabled = true
+    metric_name                = "api"
+    sampled_requests_enabled   = false
+  }
+}
+
+resource "aws_wafv2_regex_pattern_set" "body_exclusions" {
+  provider    = aws.ca-central-1
+  name        = "RequestBodyExclusions"
+  description = "Regex to match request urls with bodies that will trigger rulesets"
+  scope       = "CLOUDFRONT"
+
+  regular_expression {
+    regex_string = "^/(clamav|assemblyline)(/s3)?$"
+  }
+
+  tags = {
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+}
+
+resource "aws_wafv2_regex_pattern_set" "valid_uri_paths" {
+  provider    = aws.ca-central-1
+  name        = "ValidURIPaths"
+  description = "Regex to match the APIs valid URI paths"
+  scope       = "CLOUDFRONT"
+
+  # ops
+  regular_expression {
+    regex_string = "^/(healthcheck|version|docs|openapi.json)$"
+  }
+
+  # assemblyline
+  regular_expression {
+    regex_string = "^/assemblyline(/[\\w]{8}-[\\w]{4}-[\\w]{4}-[\\w]{4}-[\\w]{12})?$"
+  }
+
+  # clamav
+  regular_expression {
+    regex_string = "^/clamav(/([\\w]{8}-[\\w]{4}-[\\w]{4}-[\\w]{4}-[\\w]{12}|s3))?$"
+  }
+
+  tags = {
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+}
+
+resource "aws_cloudwatch_log_group" "api_waf" {
+  name              = "/aws/kinesisfirehose/api_waf"
+  retention_in_days = 14
+
+  tags = {
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+}
+
+resource "aws_kinesis_firehose_delivery_stream" "api_waf" {
+  provider = aws.ca-central-1
+
+  name        = "aws-waf-logs-simplify_privacy"
+  destination = "extended_s3"
+
+  server_side_encryption {
+    enabled = true
+  }
+
+  extended_s3_configuration {
+    role_arn           = aws_iam_role.waf_log_role.arn
+    prefix             = "waf_acl_logs/AWSLogs/154232573588/"
+    bucket_arn         = "arn:aws:s3:::${var.billing_code}-tf"
```

Comment:
This will need to be a different bucket ARN (that doesn't exist in the account yet).  It's for our Cloud Based Sensor and sends all logs it receives to the Canadian Centre for Cyber Security (CCCS).

I'll work on getting it taken care of.